azure-pipelines.yml

```diff
@@ -262,6 +262,15 @@ stages:
           artifact: RuleDescriptorBin
           targetPath: analyzers\src\SonarAnalyzer.RuleDescriptorGenerator\bin
 
+      - task: Cache@2
+        displayName: "Cache Maven local repo"
+        inputs:
+          key: 'maven | "$(Agent.OS)" | **/pom.xml'
```

Comment:
We should include the version in the key to avoid infinite grow. With help of the script you refactored recently.